python/ray/_private/parameter.py

```diff
@@ -392,3 +395,20 @@ def _check_usage(self):
         if numpy_major <= 1 and numpy_minor < 16:
             logger.warning("Using ray with numpy < 1.16.0 will result in slow "
                            "serialization. Upgrade numpy if using with ray.")
+
+    def _format_ports(self, pre_selected_ports):
+        """Format the pre selected ports information to be more
+        human readable.
+        """
+        ports = pre_selected_ports.copy()
+
+        for comp, port_list in ports.items():
+            if len(port_list) == 1:
+                ports[comp] = port_list[0]
+            elif len(port_list) == 0:
+                # Nothing is selected, meaning it will be randomly selected.
+                ports[comp] = "random"
+            elif comp == "worker_ports":
+                ports[comp] = (
+                    f"{port_list[0]}-{port_list[len(port_list) - 1]}")
```

Comment:
maybe we can rephrase it a little bit better: i.e. `f"{len(port_list) ports from {min_port} to {max_port}"`